File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandListArticle.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Telnet.TelnetClient;

public class BahamutCommandListArticle extends TelnetCommand {
    int _article_index;

    public BahamutCommandListArticle(int articleIndex) {
        this._article_index = articleIndex;
        this.Action = ListArticle;
    }

    public void execute(TelnetListPage aListPage) {
        if (this._article_index > 0) {
            TelnetClient.getClient().sendDataToServer((this._article_index + "\nS").getBytes());
        }
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[ListArticle][articleIndex=" + this._article_index + "]";
    }
}

File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandListArticle.kt
package com.kota.Bahamut.Command

import com.kota.Bahamut.ListPage.TelnetListPage
import com.kota.Bahamut.ListPage.TelnetListPageBlock
import com.kota.Telnet.TelnetClient

class BahamutCommandListArticle(var _article_index: Int) : TelnetCommand() {
    init {
        this.Action = BahamutCommandDefs.Companion.ListArticle
    }

    override fun execute(aListPage: TelnetListPage?) {
        if (this._article_index > 0) {
            TelnetClient.getClient()
                .sendDataToServer((this._article_index.toString() + "\nS").toByteArray())
        }
    }

    override fun executeFinished(aListPage: TelnetListPage?, aPageData: TelnetListPageBlock?) {
        setDone(true)
    }

    override fun toString(): String {
        return "[ListArticle][articleIndex=" + this._article_index + "]"
    }
}
